wallet-sdk/src/main/kotlin/org/stellar/walletsdk/util/FormatOperation.kt

```diff
@@ -0,0 +1,212 @@
+    // Create account
+    "create_account" -> {
+      val isCreator = (operation as CreateAccountOperationResponse).funder == accountAddress
+
+      return WalletOperation(
+        operation.id.toString(),
+        operation.createdAt,
+        operation.startingBalance,
+        account =
+          if (isCreator) {
+            operation.account
+          } else {
+            operation.funder
+          },
+        formatNativeAsset(),
+        type =
+          if (isCreator) {
+            WalletOperationType.SEND
+          } else {
+            WalletOperationType.RECEIVE
+          },
+        operation
+      )
+    }
+    // Payment
+    "payment" -> {
+      // TODO: This version of Java SDK currently doesn't have "to" and "from" muxed account for
+      // payment
+      val isSender = (operation as PaymentOperationResponse).from == accountAddress
+
+      return WalletOperation(
+        operation.id.toString(),
+        operation.createdAt,
+        operation.amount,
+        account =
+          if (isSender) {
+            operation.to
+          } else {
+            operation.from
+          },
+        formatWalletAsset(operation),
+        type =
+          if (isSender) {
+            WalletOperationType.SEND
+          } else {
+            WalletOperationType.RECEIVE
+          },
+        operation
+      )
+    }
+    // Path payment and swap
+    "path_payment_strict_receive",
+    "path_payment_strict_send" -> {
+      // TODO: check muxed account
+      operation as PathPaymentBaseOperationResponse
+      val isSender = operation.from == accountAddress
+      val isSwap = isSender && operation.from == operation.to
+
+      return WalletOperation(
+        operation.id.toString(),
+        operation.createdAt,
+        operation.amount,
+        account =
+          if (isSender) {
+            if (isSwap) {
+              ""
+            } else {
+              operation.to
+            }
+          } else {
+            operation.from
+          },
+        formatWalletAsset(operation),
+        type =
+          if (isSender) {
+            if (isSwap) {
+              WalletOperationType.SWAP
+            } else {
+              WalletOperationType.SEND
+            }
+          } else {
+            WalletOperationType.RECEIVE
+          },
+        operation
+      )
+    }
+    // Other
+    else -> {
+      return WalletOperation(
+        operation.id.toString(),
+        operation.createdAt,
+        "",
+        "",
+        listOf(),
+        WalletOperationType.OTHER,
+        operation
+      )
+    }
+  }
+}
+
+/**
+ * Format anchor transactions to make them consistent.
+ *
+ * @param transaction anchor transaction to format
+ * @param asset transaction asset
+ *
+ * @return formatter transaction
+ */
+fun formatAnchorTransaction(
+  transaction: AnchorTransaction,
+  asset: Asset
+): WalletOperation<AnchorTransaction> {
+  val formattedAsset = formatAsset(asset)
+
+  when (transaction.kind) {
```

Comment:
out of scope of this PR probably, but I think transaction.kind should be a enum

Reply:
:+1: Will add it to the list.